packages/components/src/components/modal/modal.tsx

```diff
@@ -109,6 +111,12 @@ export class Modal {
     }
   }
 
+  closeEvent(type: string) {
+    if (!this.scaleBeforeClose.emit(type).defaultPrevented) {
+      this.opened = false;
+    }
+  }
```

Comment:
I was thinking I might have made up the "convention" thing because I didn't do any research within the project to confirm it (that we use objects in `event.detail`, so it might be more a personal thing 😁

Reply:
Putting Objects into the details sounds rational since you can add other stuff later without breaking anything.

The rename is also good.

Reply:
I hadn't thought about the non-breaking outcome, now I like it even more. Anything "non-breaking" is good for us at this point :)